esp/services/esdl_svc_engine/esdl_binding.cpp

```diff
@@ -363,20 +363,7 @@ static void ensureMergeOrderedEsdlTransform(Owned<IPropertyTree> &dest, IPropert
     {
         Owned<IPropertyTreeIterator> children = copy->getElements("*");
         ForEach(*children)
-        {
-            const char *existing = children->query().queryProp("@target");
-            if (existing && *existing)
-            {
-                if (*existing=='.') //support in future so error now so we don't have backward compatability issues
-                {
-                    VStringBuffer errorXpath("unsupported choose xpath*>>%s", existing); //will cause request time xpath error
-                    children->query().setProp("@target", errorXpath);
-                    DBGLOG("ESDL Binding: request transform choose xpath error: %s", existing);
-                }
-                continue;
-            }
-            children->query().setProp("@target", target);
-        }
+            children->query().setProp("@_crtTarget", target); //internal use only attribute
```

Comment:
I think the check for existing is still necessary. A top level choose that specifies a custom target will be overwritten, but descendant instances will be honored (whether or not the attribute should be there).

My only issue with the using `@target` was the future possibility adding this value to an element that uses that name for something else.

Reply:
This happens at load time, the usage and passing the target tree to descendants happens at run time.  This just puts the target in the right place.  The reason I checked for existing before was because I was letting the user set a target at "choose".  I'm not supporting that now.  That's a future enhancement.

Reply:
So this only pushes an internal attribute down.  There will be no existing because the user isn't supposed to add a _crtTarget attribute on there own.  If they do it will be overwritten as expected.